indico/modules/events/registration/wallets/apple.py

```diff
@@ -0,0 +1,144 @@
+        self.cert = None
+
+    @property
+    def is_configured(self):
+        return config.ENABLE_APPLE_WALLET and self.settings is not None
+
+    def build_ticket_object(self, registration):
+        from indico.modules.categories.controllers.util import make_format_event_date_func
+        ticket = EventTicket()
+        ticket.addPrimaryField('event-title', self.event.title, _('Event'))
+        event_date = make_format_event_date_func(self.event.category)(self.event)
+        ticket.addSecondaryField('event-date', event_date, _('Date'))
+        if self.event.address and self.event.has_location_info:
+            venue_name = (f'{self.event.room_name} ({self.event.venue_name})'
+                          if self.event.room_name and self.event.venue_name
+                          else (self.event.venue_name or self.event.room_name))
+            ticket.addSecondaryField('event-venue', venue_name, _('Venue'))
+            ticket.addBackField('back-event-venue', venue_name, _('Venue'))
+        ticket.addAuxiliaryField('registration-name', registration.full_name, _('Name'))
+        ticket.addAuxiliaryField('registration-email', registration.email, _('Email'))
+        ticket.addBackField('back-registration-name', registration.full_name, _('Name'))
+        ticket.addBackField('back-ticket-number', f'#{registration.friendly_id}', _('Ticket number'))
+        ticket.addBackField('back-event-date', event_date, _('Date'))
+        ticket.addBackField('back-registration-email', registration.email, _('Email'))
+        ticket.addBackField('back-event-url', f'<a href="{self.event.external_url}">{_("Event page")}</a>', _('Link'))
+        reg_url = url_for('event_registration.display_regform', registration.locator.uuid, _external=True)
+        ticket.addBackField('back-registration-details', f'<a href="{reg_url}">{_("Registration details")}</a>',
+                            _('Link'))
+
+        signals.event.registration.apple_wallet_ticket_object.send(self.event, obj=ticket)
+        return ticket
+
+    def build_pass_object(self, registration):
+        from indico.modules.events.registration.util import get_persons, get_ticket_qr_code_data
+
+        # Extract Certificate details from certificate.pem itself
+        self.cert = x509.load_pem_x509_certificate(self.settings['apple_wallet_certificate'].encode())
+        cert_details = {d.split('=')[0]: d.split('=')[1] for d in self.cert.subject.rfc4514_string().split(',')}
+        passfile = IndicoPass(self.build_ticket_object(registration), passTypeIdentifier=cert_details['UID'],
+                              organizationName=cert_details['O'], teamIdentifier=cert_details['OU'])
+        passfile.logoText = cert_details['O']  # Add Organization name at the top
+        passfile.backgroundColor = '#007cac'
+        passfile.foregroundColor = passfile.labelColor = '#ffffff'
+        passfile.serialNumber = registration.ticket_uuid
```

Comment:
just to be clear that the saved serial from the DB has priority in case we ever need to change it for a registration:

```suggestion
        passfile.serialNumber = registration.apple_wallet_serial or registration.ticket_uuid
```

---

also, since you currently *always* use the ticket UUID for this, I wonder if it'd make more sense to:

- just always use the `ticket_uuid` instead of using a separate column
- use a different UUID which is generated the first time the applet wallet is created for a registration

just a thought though, i don't have a strong opinion on doing either of these two things so far

Reply:
I think that keeping the `apple_wallet_serial` will allow us (and anyone else) to customize the serialNumber easily based on specific needs, so I suggest to keep it.
For our need, at the moment, the ticket_uuid will work so I suggest to keep it as is.
Pls, lemme know.

Reply:
Yes, I'm fine with keeping the column, and let's keep copying the `ticket_uuid` in that case. I think the suggestion made in [this comment](https://github.com/indico/indico/pull/6248#discussion_r1601390389) still makes sense though.